DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/src/ddr_hosts_model.cpp

```diff
@@ -0,0 +1,115 @@
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+#include <sycl/ext/oneapi/annotated_arg/annotated_ptr.hpp>
+
+#include "exception_handler.hpp"
+
+using namespace sycl;
+using namespace ext::oneapi::experimental;
+
+using usm_buffer_location =
+    ext::intel::experimental::property::usm::buffer_location;
+
+constexpr int kBL1 = 1;
+constexpr int kBL2 = 2;
+
+struct DDRIP{
+  annotated_ptr<int, decltype(properties{
+          buffer_location<kBL1>,
+          maxburst<8>,
+          dwidth<256>,
+          alignment<32>
+          })> x; 
+  annotated_ptr<int, decltype(properties{
+          buffer_location<kBL1>,
+          maxburst<8>,
+          dwidth<256>,
+          alignment<32>
+          })> y;
+  annotated_ptr<int, decltype(properties{
+          buffer_location<kBL2>,
+          maxburst<8>,
+          dwidth<256>,
+          alignment<32>
+          })> z;  
+
+  int size;
+
+  void operator()() const {
+    for(int i = 0; i < size; i++){
+      int add = x[i] + y[i];
+      z[i] = add;
+    }
+  }
+};
+
+int main(void){
+#if FPGA_SIMULATOR
+  auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+  auto selector = sycl::ext::intel::fpga_selector_v;
+#else  // #if FPGA_EMULATOR
+  auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+  bool passed = true;
+
+  try {
+    // create the device queue
+    sycl::queue q(selector, fpga_tools::exception_handler,
+                  sycl::property::queue::enable_profiling{});
+
+    // make sure the device supports USM host allocations
+    sycl::device device = q.get_device();
+
+    // Print out the device information.
+    std::cout << "Running on device: "
+              << q.get_device().get_info<sycl::info::device::name>().c_str()
+              << std::endl;
+
+    if (!device.has(sycl::aspect::usm_host_allocations)) {
+      std::terminate();
+    }
+
+    // Create and initialize the host arrays
+    constexpr int kN = 8;
+    std::cout << "Elements in vector : " << kN << "\n";
+
+    int *array_A = malloc_shared<int>(kN, q, property_list{usm_buffer_location(kBL1)});
+    int *array_B = malloc_shared<int>(kN, q, property_list{usm_buffer_location(kBL1)});
+    int *array_C = malloc_shared<int>(kN, q, property_list{usm_buffer_location(kBL2)});
+
+    for(int i = 0; i < kN; i++){
+        array_A[i] = i;
+        array_B[i] = 2*i;
+    }
+
+    q.single_task(DDRIP{array_A, array_B, array_C, kN}).wait();
+    for (int i = 0; i < kN; i++) {
+      auto golden = 3*i;
+      if (array_C[i] != golden) {
+        std::cout << "ERROR! At index: " << i << " , expected: " << golden
+                  << " , found: " << array_C[i] << "\n";
+        passed = false;
+      }
+    }
+
+    free(array_A, q);
+    free(array_B, q);
+    free(array_C, q);
+  }
+  catch(sycl::exception const &e){
```

Comment:
I don't think we need to do exception catching blocks anymore. Check with Yohann and then remove it.

Reply:
@whitepau Yohann is out of office, do we need to do exception catching blocks? hostpipes and device_global tutorials has it currently